src/pyscheduling/Problem.py

```diff
@@ -37,6 +42,200 @@ def __lt__(self, other):
         """
         return self.name < other.name
 
+class Constraints():
+
+    @staticmethod
+    def sorting_func(constraint):
+        if constraint is Constraints.W : return 0
+        elif constraint is Constraints.R : return 1
+        elif constraint is Constraints.S : return 2
+        elif constraint is Constraints.D : return 3
+
+    class W():
+        @staticmethod
+        def create_attribute(cls,var):
+            setattr(cls,"W",var) if var is not None else setattr(cls,"W",list())
+
+        @staticmethod
+        def read_attribute(cls, text_content : list[str], starting_index : int):
+            W, i = Instance.read_1D(text_content,starting_index)
+            setattr(cls,"W",W)
+            return i
+            
+        @staticmethod
+        def print_attribute(cls, file):
+            W = getattr(cls,"W")
+            file.write("\nWeights\n")
+            for i in range(cls.n):
+                file.write(str(W[i])+"\t")
+
+        @staticmethod
+        def generate_attribute(**kwargs):
+            law = kwargs.get("law")
+            instance = kwargs.get("instance")
+            Wmin = kwargs.get("Wmin")
+            Wmax = kwargs.get("Wmax")
+            W = []
+            for j in range(instance.n):
+                if law.name == "UNIFORM":  # Generate uniformly
+                    n = int(random.randint(Wmin, Wmax+1))
+                elif law.name == "NORMAL":  # Use normal law
+                    value = np.random.normal(0, 1)
+                    n = int(abs(Wmin+Wmax*value))
+                    while n < Wmin or n > Wmax:
+                        value = np.random.normal(0, 1)
+                        n = int(abs(Wmin+Wmax*value))
+                W.append(n)
+
+            instance.W = W
+            return W
+
+    class R():
+        @staticmethod
+        def create_attribute(cls,var):
```

Comment:
Since cls is always the instance, why not name it instance ? it's much clearer that way

Reply:
you're right